### ISSUE TITLE
Rerunning a group_module task fails on OS X

### ISSUE THREAD

**Issue by anthonygreen** [ansible/ansible-modules-core#1923]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.9.2
##### Environment:

Running from: OS X 10.95
Managing: OS X 10.10.2
##### Summary:

With OSX Ansible group_module requires the gid flag (see #1922) but adding it generates an error on task reruns
##### Steps To Reproduce:

Run the Ansible task to create a group in OS X

```
- name: Add jenkins group
  group: name=jenkins gid=500
  sudo: yes
```

now rerunning it, even with gid flag removed results in a failure

```
- name: Add jenkins group
  group: name=jenkins
  sudo: yes
```

failed: [jenkins_slave_osx] => {"failed": true, "parsed": false}

BECOME-SUCCESS-ibsurliyaklkntgblhlxixwdnxzmwovc
Traceback (most recent call last):
  File "/Users/mediaplayout/.ansible/tmp/ansible-tmp-1439807405.6-11720088341903/group", line 2051, in <module>
    main()
  File "/Users/mediaplayout/.ansible/tmp/ansible-tmp-1439807405.6-11720088341903/group", line 423, in main
    (rc, out, err) = group.group_mod(gid=group.gid)
TypeError: group_mod() got an unexpected keyword argument 'gid'

**Comment by thudry** [comment]
This is fixed in the devel
#1086

**Comment by bcoca** [comment]
fix was applied to stable-1.9 so it should be included in 1.9.3 also
